src/Types/JsonWsp.purs

```diff
@@ -153,6 +188,37 @@ parseMirror = caseJsonObject (Left (TypeMismatch "expected object")) $
       pure { step, id }
   )
 
+newtype Assets = Assets (Map CurrencySymbol (Map TokenName BigInt))
+
+instance DecodeJson Assets where
+  decodeJson j = do
+    wspAssets :: Array (String /\ String) <- FO.toUnfoldable <$> decodeJson j
+    Assets <<< Map.fromFoldableWith Map.union <$> sequence (uncurry decodeAsset <$> wspAssets)
```

Comment:
Could I just confirm we're relying on duplicate keys to be impossible for JSON here? Because `Map.union` has a left bias on duplicate keys and wanted to make sure we aren't "throwing away" `BigInt` (for a given CurrencySymbol and duplicate TokenNames) which is fine if `"assets"` fields are indeed unique.

Reply:
Duplicated keys - a.k.a. more than one entry for the same pair of currencysymbol and TokenName will cause data loss you mentioned above.
Should I replace it with `unionWith (+)` ?

Reply:
I think so yes, just to be safe.